src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -106,24 +131,132 @@ export default class SquarePaymentStrategy extends PaymentStrategy {
                 unsupportedBrowserDetected: () => {
                     deferred.reject(new UnsupportedBrowserError());
                 },
-                cardNonceResponseReceived: (errors, nonce) => {
-                    this._cardNonceResponseReceived(errors, nonce);
+                cardNonceResponseReceived: (errors: NonceGenerationError[] | undefined, nonce: string | undefined, cardData: CardData | undefined,
```

Comment:
I might be wrong, but you should be able to replace the `| undefined` with a `?` before the `:`:

```js
cardNonceResponseReceived: (errors?: NonceGenerationError[], nonce?: string, cardData?: CardData, billingContact?: Contact, shippingContact?: Contact) => {
```

also, if you want to split the line, you'll have to put all parameters on new lines:

```js
cardNonceResponseReceived: (
	errors?: NonceGenerationError[], 
	nonce?: string, 
	cardData?: CardData, 
	billingContact?: Contact, 
	shippingContact?: Contact
) => {
```

Since you have a typedef for this callbacks, shouldn't just try to use those?

and also, since you are not using most of the data in this callback, you shouldn't need to put it in the signature of the function.

Reply:
ng-checkout does.